clients/src/main/java/org/apache/kafka/clients/KafkaClient.java

```diff
@@ -81,6 +81,14 @@
     public List<ClientResponse> poll(long timeout, long now);
 
     /**
+     * Disconnect connection to a particular node (if there is one). The disconnection is asynchronous and will not be
+     * processed until the next {@link #poll(long, long) poll()} call.
```

Comment:
Do we need to wait for the next poll() call? disconnect() will cancel the key, after which the key will never be selected again.